elastica/joint.py

```diff
@@ -222,51 +228,68 @@ def __init__(self, k, nu, kt):
-        # also NOTE: - rod two is fixed at first element
-        link_direction = (
-            rod_two.position_collection[..., index_two + 1]
-            - rod_two.position_collection[..., index_two]
+    def apply_torques(self, system_one, index_one, system_two, index_two):
+        # collect directors of systems one and two
+        # note that systems can be either rods or rigid bodies
+        system_one_director = system_one.director_collection[..., index_one]
+        system_two_director = system_two.director_collection[..., index_two]
+
+        if self.static_rotation is None:
+            # this if clause should be active during the first timestep for the case use_static_rotation==True
+            self.static_rotation = system_one_director @ system_two_director.T
+
+        # rel_rot: C_12 = C_1I @ C_I2
+        # C_12 is relative rotation matrix from system 1 to system 2
+        # C_1I is the rotation from system 1 to the inertial frame (i.e. the world frame)
+        # C_I2 is the rotation from the inertial frame to system 2 frame (inverse of system_two_director)
+        rel_rot = system_one_director @ system_two_director.T
+        # error_rot: C_22* = C_21 @ C_12*
+        # C_22* is rotation matrix from current orientation of system 2 to desired orientation of system 2
+        # C_21 is the inverse of C_12, which describes the relative (current) rotation from system 1 to system 2
+        # C_12* is the desired rotation between systems one and two, which is saved in the static_rotation attribute
+        error_rot = rel_rot.T @ self.static_rotation
+
+        # compute rotation vectors based on C_22*
+        rot_vec = Rotation.from_matrix(error_rot).as_rotvec()
+
+        # rotate rotation vector into inertial frame
+        rot_vec_inertial_frame = system_two_director.T @ rot_vec
+
+        # error in rotation velocity between system 1 and system 2
+        error_omega = (
+            system_one.omega_collection[..., index_one]
+            - system_two.omega_collection[..., index_two]
         )
 
-        # To constrain the orientation of link two, the second node of link two should align with
-        # the direction of link one. Thus, we compute the desired position of the second node of link two
-        # as check1, and the current position of the second node of link two as check2. Check1 and check2
-        # should overlap.
-
-        tgt_destination = (
-            rod_one.position_collection[..., index_one]
-            + rod_two.rest_lengths[index_two] * rod_one.tangents[..., index_one]
-        )  # dl of rod 2 can be different than rod 1 so use rest length of rod 2
-
-        curr_destination = rod_two.position_collection[
-            ..., index_two + 1
-        ]  # second element of rod2
+        # we compute the constraining torque using a rotational spring - damper system in the inertial frame
+        torque = self.kt * rot_vec_inertial_frame - self.nut * error_omega
```

Comment:
```suggestion
        torque = self.kt * rot_vec_inertial_frame - self.nut * dev_omega
```

Reply:
Resolved in https://github.com/GazzolaLab/PyElastica/pull/135/commits/3b7a1fdbebe2b8fa3c9b1a6430a25defed70e746